### PR TITLE
Using try_run to get __MAC_OS_X_VERSION_MAX_ALLOWED

### PR DESCRIPTION
**Pull request by MatthiasWM**
Possible solution for #1103 ?

### CODE REVIEW

File: CMake/setup.cmake [hunk 1]
```diff
@@ -117,6 +117,17 @@ if(APPLE)
   set(HAVE_VSNPRINTF 1)
   set(HAVE_SCANDIR 1)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated")
+  try_run(
+    RUN_RESULT COMPILE_RESULT
+    SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/CMake/macosMaxAllowd.c
+    RUN_OUTPUT_VARIABLE MAC_OS_X_VERSION_MAX_ALLOWED
+  )
+  if (COMPILE_RESULT STREQUAL "FALSE")
```
Comment (Albrecht-S) [inline CMake/setup.cmake:125]: Can probably be simplified to:
```
if (NOT COMPILE_RESULT)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -345,3 +345,5 @@ if(0) # debug built library and fluid targets
   endforeach()
   message(STATUS "---------------------------------------------------------")
 endif()
+
```
Comment (Albrecht-S) [inline CMakeLists.txt:348]: Adding these two empty lines should be reverted. This file shouldn't be changed at all

File: CMake/macosMaxAllowd.c [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#include <AvailabilityMacros.h>
+#include <stdio.h>
+
+int main(int argc, char **argv) {
+  unsigned int ver = __MAC_OS_X_VERSION_MAX_ALLOWED;
+  printf("%d.%d.%d", ver/10000, (ver/100)%100, ver%100);
+  return 0;
+}
```
Comment (Albrecht-S) [inline CMake/macosMaxAllowd.c:8]: If this is going to be used, please rename this to `CMake/macosMaxAllowed.c` (fix spelling).

File: CMake/macOSMaxAllowed.c [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#include <AvailabilityMacros.h>
```
Comment (Albrecht-S) [inline CMake/macOSMaxAllowed.c:1]: @MatthiasWM, @ManoloFLTK Please check if this header file is available on all targeted macOS versions. ISTR that the header `<Availability.h>` was used on older macOS versions.

File: CMake/macOSMaxAllowed.c [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+#include <stdio.h>
+
+int main(int argc, char **argv) {
+  unsigned int ver = __MAC_OS_X_VERSION_MAX_ALLOWED;
+  printf("%d.%d.%d", ver/10000, (ver/100)%100, ver%100);
```
Comment (Albrecht-S) [inline CMake/macOSMaxAllowed.c:6]: Same as above, should it fail if the "underscored" macro is not defined?

File: CMake/macOSMaxAllowed.c (hunk 4 above)
Comment (Albrecht-S) [reply CMake/macOSMaxAllowed.c:1]: _Hmm, I'm confused, I thought I added this already:_

Maybe it's intended to fail if `<AvailabilityMacros.h>` does not exist? Then it's correct, obviously.

Comment (MatthiasWM) [reply CMake/macOSMaxAllowed.c:1]: `<Availability.h>` is the correct header, and `__MAC_OS_X_VERSION_MAX_ALLOWED` should always be defined there if we target OS X (vs. iPhone, watch, or what have you). If the include file does not exist, or the macro is not there, the caller falls back to the host system version `CURRENT_OSX_VERSION`.

File: CMake/macOSMaxAllowed.c (hunk 5 above)
Comment (MatthiasWM) [reply CMake/macOSMaxAllowed.c:6]: Fixed the include file, failure calls a reasonable fallback. Tested.

**Pull request merged**